spark-bigquery-pushdown/pushdown_common_src/main/scala/com/google/cloud/spark/bigquery/pushdowns/UnionQuery.scala

```diff
@@ -0,0 +1,71 @@
+package com.google.cloud.spark.bigquery.pushdowns
+
+import com.google.cloud.spark.bigquery.pushdowns.SparkBigQueryPushdownUtil.{blockStatement, makeStatement}
+import org.apache.spark.sql.catalyst.expressions.{Attribute, AttributeReference, NamedExpression}
+import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan
+
+import scala.Seq
+
+case class UnionQuery(
+                  expressionConverter: SparkExpressionConverter,
+                  expressionFactory: SparkExpressionFactory,
+                  sparkPlanFactory: SparkPlanFactory,
+                  children: Seq[BigQuerySQLQuery],
+                  outputAttributes: Option[Seq[Attribute]],
+                  alias: String)
+  extends BigQuerySQLQuery(
+    expressionConverter,
+    expressionFactory,
+    alias,
+    children = children,
+    outputAttributes = outputAttributes)  {
+  
+  override val columnSet: Seq[Attribute] = {
```

Comment:
We will need to pass `visibleAttribute` to the parent `BigQuerySQLQuery` class. If we don't, then the parent class will be constructed first with the incorrect `columnSet`. As a result, all other variables in the parent class that use `columnSet` will be constructed incorrectly as well

Reply:
Addressed